python-lib/dku_utils/node_pool.py

```diff
@@ -106,7 +106,7 @@ def build_node_pool_taints_yaml(node_pool):
     yaml_taints = []
     if node_pool['taints']:
         for taint in node_pool['taints']:
-            if not _is_none_or_blank(taint.get('key', '')):
+            if not _is_none_or_blank(taint.get('key', None)):
```

Comment:
Since we are the only ones working on this, I'd rather remove Python 2 support.
We do not have to add any specific features in the close future, so we could focus on that instead.